pkg/actors/actors.go

```diff
@@ -1103,15 +1102,35 @@ func (a *actorsRuntime) doExecuteReminderOrTimerOnInternalActor(ctx context.Cont
 			return err
 		}
 	} else {
-		log.Debugf("Executing reminder for internal actor '%s'", reminder.Key())
+		key := reminder.Key()
+
+		log.Debugf("Executing reminder for internal actor '%s'", key)
+
+		a.lock.Lock()
+		if _, ok := a.internalReminderInProgress[key]; ok {
+			a.lock.Unlock()
+			// We don't need to return cancel here as the first invocation will
+			// delete the reminder.
+			log.Debugf("Duplicate concurrent reminder invocation detected for '%s', likely due to long processing time. Ignoring in favour of the active invocation", key)
```

Comment:
This scares me.

Reply:
We could gate this map check behind the scheduler actor reminder feature gate?

> Is there a bug in how scheduler integrates into reminders that causes this abstraction leak? Firing reminders always had single thread guarantees in Dapr.

> Is there a bug in how scheduler integrates into reminders that causes this abstraction leak? Firing reminders always had single thread guarantees in Dapr.

We are only dealing with internal actor reminders here, which as generally implemented today, are a bit of a hack anyway w.r.t. how actors works generally in application code.